shared/src/price_estimation/paraswap.rs

```diff
@@ -35,28 +35,33 @@ impl ParaswapPriceEstimator {
         ensure_token_supported(query.buy_token, self.bad_token_detector.as_ref()).await?;
         ensure_token_supported(query.sell_token, self.bad_token_detector.as_ref()).await?;
 
-        let (src_token, dest_token, side) = match query.kind {
-            OrderKind::Buy => (query.buy_token, query.sell_token, Side::Buy),
-            OrderKind::Sell => (query.sell_token, query.buy_token, Side::Sell),
-        };
         let price_query = PriceQuery {
-            src_token,
-            dest_token,
-            src_decimals: decimals(&src_token, token_infos)? as usize,
-            dest_decimals: decimals(&dest_token, token_infos)? as usize,
+            src_token: query.sell_token,
+            dest_token: query.buy_token,
```

Comment:
This is a drive by fix because while testing I noticed I was using src and dest token wrong in this file.